src/lib.rs

```diff
@@ -46,31 +60,673 @@ pub enum ListStatus {
+        require!(!payments.is_empty(), "Payment list cannot be empty");
+
+        let caller = env::predecessor_account_id();
+
+        // Verify storage credits
+        let required_credits = payments.len() as u128;
+        let current_credits = self
+            .storage_credits
+            .get(&caller)
+            .copied()
+            .unwrap_or(NearToken::from_yoctonear(0))
+            .as_yoctonear();
+
+        require!(
+            current_credits >= required_credits,
+            format!(
+                "Insufficient storage credits. Required: {}, Available: {}",
+                required_credits, current_credits
+            )
+        );
+
+        // Deduct storage credits
+        let new_credits = NearToken::from_yoctonear(current_credits - required_credits);
+        self.storage_credits.insert(caller.clone(), new_credits);
+
+        // Convert PaymentInput to PaymentRecord with Pending status
+        let payment_records: Vec<PaymentRecord> = payments
+            .into_iter()
+            .map(|input| PaymentRecord {
+                recipient: input.recipient,
+                amount: input.amount.0,
+                status: PaymentStatus::Pending,
+            })
+            .collect();
+
+        // Create payment list
+        let list_id = self.next_list_id;
+        self.next_list_id += 1;
+
+        let payment_list = PaymentList {
+            token_id,
+            submitter: caller.clone(),
+            status: ListStatus::Pending,
+            payments: payment_records,
+            created_at: env::block_timestamp(),
+        };
+
+        self.payment_lists.insert(list_id, payment_list);
+
+        log!(
+            "Payment list {} submitted by {} with {} payments",
+            list_id,
+            caller,
+            self.payment_lists.get(&list_id).unwrap().payments.len()
+        );
+
+        list_id
+    }
+
+    /// Approve a payment list and attach the exact deposit amount
+    #[payable]
+    pub fn approve_list(&mut self, list_ref: u64) {
+        let caller = env::predecessor_account_id();
+
+        let mut list = self
+            .payment_lists
+            .get(&list_ref)
+            .expect("Payment list not found")
+            .clone();
+
+        require!(
+            list.submitter == caller,
+            "Only the submitter can approve the list"
+        );
+
+        require!(
+            matches!(list.status, ListStatus::Pending),
+            "List must be in Pending status"
+        );
+
+        // Calculate total payment amount (with overflow check)
+        let total_amount: u128 = list
+            .payments
+            .iter()
+            .map(|p| p.amount)
+            .try_fold(0u128, |acc, x| acc.checked_add(x))
+            .expect("Total payment amount overflow");
+
+        let attached = env::attached_deposit();
+        let required = NearToken::from_yoctonear(total_amount);
+
+        require!(
```

Comment:
For Fungible Token payment lists, the approval must go via `ft_on_transfer`, and the caller must call `ft_transfer_call`with the correct token and amount to be able to approve. See https://docs.near.org/primitives/ft/standard#ft_transfer_call